libcontainer/dmz/dmz_linux.go

```diff
@@ -43,9 +43,9 @@ var (
 // If the runc-dmz binary is not embedded into the runc binary, Binary will
 // return ErrNoDmzBinary as the error.
 func Binary(tmpDir string) (*os.File, error) {
-	// Setting RUNC_DMZ=legacy disables this dmz method.
-	if os.Getenv("RUNC_DMZ") == "legacy" {
-		logrus.Debugf("RUNC_DMZ=legacy set -- switching back to classic /proc/self/exe cloning")
+	// Only RUNC_DMZ=true enables runc_dmz
+	if os.Getenv("RUNC_DMZ") != "true" {
```

Comment:
Can we use strconv.ParseBool

Reply:
@AkihiroSuda it returns an error which we need to handle and also errors out when parsing `""`. So, the code becomes a mess for, IMHO, no good reason:

```
-       // Setting RUNC_DMZ=legacy disables this dmz method.
-       if os.Getenv("RUNC_DMZ") == "legacy" {
-               logrus.Debugf("RUNC_DMZ=legacy set -- switching back to classic /proc/self/exe cloning")
+       // Only RUNC_DMZ=true enables runc_dmz
+       runcDmz := os.Getenv("RUNC_DMZ")
+       if runcDmz == "" {
+               logrus.Debugf("RUNC_DMZ is not set -- switching back to classic /proc/self/exe cloning")
                return nil, ErrNoDmzBinary
        }
+       if dmzEnabled, err := strconv.ParseBool(runcDmz); err == nil && !dmzEnabled {
+               logrus.Debugf("RUNC_DMZ is not true -- switching back to classic /proc/self/exe cloning")
+               return nil, ErrNoDmzBinary
+       } else if err != nil {
+               return nil, fmt.Errorf("parsing RUNC_DMZ: %w", err)
+       }
+
```

Do you really think it is worth it?

Reply:
Yes, to allow `RUNC_DMZ=1`

Reply:
Updated to do that!